fast64_internal/f3d/f3d_parser.py

```diff
@@ -690,16 +730,20 @@ def addTriangle(self, indices, dlData):
         for i in range(int(len(indices) / 3)):
             self.triMatIndices.append(self.lastMaterialIndex)
 
+    # Add items to this tuple in child classes
+    def getMaterialKey(self, material: bpy.types.Material):
+        return material.f3d_mat.key()
+
     def getMaterialIndex(self):
-        # We do this for now to update tile settings for S and T.
-        # Right now we don't handle those, so we need the auto-calculator to set it correctly.
-        overrideContext = bpy.context.copy()
-        overrideContext["material"] = self.materialContext
-        bpy.ops.material.update_f3d_nodes(overrideContext)
 
-        for material in self.materials:
-            if propertyGroupEquals(self.materialContext.f3d_mat, material.f3d_mat):
+        key = self.getMaterialKey(self.materialContext)
+        if key in self.materialDict:
+            material = self.materialDict[key]
```

Comment:
`materialContext` is just a regular blender material. Whenever a new material needs to be made, a copy is created and stored in a list. If a matching is key is detected, then the existing material in that list is used instead of creating a new copy.